pkg/kubelet/kubeletconfig/util/files/files.go

```diff
@@ -63,34 +63,53 @@ func EnsureFile(fs utilfs.Filesystem, path string) error {
 	return file.Close()
 }
 
-// ReplaceFile replaces the contents of the file at `path` with `data` by writing to a tmp file in the same
-// dir as `path` and renaming the tmp file over `path`. The file does not have to exist to use ReplaceFile.
-func ReplaceFile(fs utilfs.Filesystem, path string, data []byte) error {
+// WriteTmpFile creates a temporary file at `path`, writes `data` into it, and fsyncs the file
+func WriteTmpFile(fs utilfs.Filesystem, path string, data []byte) (tmpPath string, err error) {
 	dir := filepath.Dir(path)
 	prefix := filepath.Base(path)
 
 	// create the tmp file
 	tmpFile, err := fs.TempFile(dir, prefix)
 	if err != nil {
-		return err
+		return "", err
 	}
+	defer func() {
+		// sync file, to ensure it's written in case a hard reset happens
+		if syncErr := tmpFile.Sync(); syncErr != nil && err == nil {
```

Comment:
I think this is fine but unnecessary - sync can just be called after write if there was no error. If write panics for some reason we don't need to call sync.
Close and Remove 👍

Reply:
Hm, that made me think - we should probably remove the temporary file if there was a panic and then re-panic. WDYT? I don't know if this is an important place in code so this may be overkill.

Reply:
Done.